Fix(anta_runner): Update Ansible task failed criteria and various logging issues (#5341)
aristanetworks/ansible-avd: ansible_collections/arista/avd/roles/anta_runner/README.md

The artifacts generated by the role, including results from ANTA, will be stored in the `anta` folder in the inventory directory.

The `Run ANTA on EOS devices` task will fail if any of the following conditions occur:

- **No tests executed:** If no tests were run against any device (and it's not a `dry_run`)

> - **Execution errors:** If errors arise during the role's execution, originating from Ansible, PyAVD, or ANTA

Reviewer: ```suggestion
- **Execution errors:** If errors arise during the role execution, originating from Ansible, PyAVD, or ANTA
```

- **Test failures or errors:** If any ANTA test reports a failure or error status.

!!! tip

>     Standard Ansible error handling (e.g., `ignore_errors`) can be used if desired. For more details, refer to the [Ansible documentation](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html).

Reviewer: ```suggestion
    To continue running the playbook when `anta_runner` fails, use Ansible `ignore_errors`. For more details, refer to the [Ansible documentation](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html).
```

# Run ANTA in dry-run mode. This will generate the tests but not execute them.
anta_runner_dry_run: false

# When `true`, the plugin returns `failed: true` if any ANTA tests fail or error.
# Otherwise, the plugin returns `changed: true` in such cases.

> anta_strict_mode: false

Reviewer: I think the readme should state somewhere that the task will fail if there is any error

Reviewer: it does at the end but it should not be at the end :)

Author: Also add a tip about ignore_errors if they want.

Author: Done.

- VerifyTransceiversTemperature
- VerifyZeroTouch

## Logging and Troubleshooting

This section details how to control log verbosity, understand log output, and diagnose common issues for the `anta_runner` role.

### Controlling Log Verbosity

> The `anta_runner` role utilizes Ansible's standard verbosity flags (`-v`, `-vv`, etc.) to control log output from its various components, including PyAVD, ANTA, underlying libraries, and the `anta_workflow` action plugin. Adjust the verbosity level using the `ansible-playbook` command:

Reviewer: ```suggestion
The `anta_runner` role uses Ansible standard verbosity flags (`-v`, `-vv`, etc.) to control log output from its various components, including PyAVD, ANTA, underlying libraries, and the `anta_workflow` action plugin. Adjust the verbosity level using the `ansible-playbook` command:
```

Author: Fixed.

- **Default (no `-v` flag):** `WARNING` level and above for all sources

> - `-v`: `INFO` level for all sources, except `httpx` logs (which remain at `WARNING` due to high chattiness)

Reviewer: httpx as WARNING?

Author: As discussed, will remove the `-v` descriptions.

Author: Removed.

- `-vv`: `INFO` level for all sources, including `httpx`
- `-vvv`: `DEBUG` level for all sources except ANTA **and** its low-level libraries

> - `-vvvv`: `DEBUG` level for all sources except ANTA low-level libraries

Reviewer: this does not make sense

Author: Removed the all `-v` details.

- `-vvvvv`: `DEBUG` level for all sources

!!! tip

>     When troubleshooting why specific AVD-generated tests might be missing from a device's final catalog, running `ansible-playbook` with `-vvv` is particularly helpful. PyAVD will output detailed information about skipped tests and the reasons for their exclusion.

Reviewer: ```suggestion
    Running `ansible-playbook` with `-vvv` is particularly helpful to troubleshoot why some specific AVD-generated tests are missing from a device final catalog.  PyAVD will output detailed information about skipped tests and the reasons for their exclusion.
```

Author: Fixed.

### Understanding Log Output

`anta_runner` processes devices in parallel batches for improved performance. To help correlate log messages to specific operations or device batches, each log entry is prefixed with a unique identifier:

- `anta-workflow`: Indicates logs from the main Ansible action plugin process.
- `anta-run-xxxxxxxx`: Indicates logs from a specific ANTA execution run handling a batch of devices.

These prefixes are essential when debugging issues related to a particular device or test batch.

### Managing Log Files

> Console logs adhere to standard Ansible logging configurations. For persistent storage and easier analysis, especially with high verbosity, consider configuring `log_path` in your `ansible.cfg` file. This redirects the primary output stream to a specified log file, which is very useful for troubleshooting. For more details, refer to the [Ansible Logging Documentation](https://docs.ansible.com/ansible/latest/reference_appendices/logging.html).

Reviewer: ```suggestion
Console logs adhere to standard Ansible logging configurations. For persistent storage and easier analysis, especially when using high verbosity, consider configuring `log_path` in your `ansible.cfg` file to redirect the primary output stream to the specified log file. For more details, refer to the [Ansible Logging Documentation](https://docs.ansible.com/ansible/latest/reference_appendices/logging.html).
```

Author: Fixed.

### Task Failure Conditions

The `anta_runner` role will mark the Ansible task as failed if any of the following conditions occur:

- **No tests executed:** If no tests were run against any device (and it's not a `dry_run`)
- **Execution errors:** If errors arise during the role's execution, originating from Ansible, PyAVD, or ANTA
- **Test failures or errors:** If any ANTA test reports a failure or error status.

> Understanding these conditions can help you quickly identify the cause of a failed playbook run.

Reviewer: This needs to be higher up in the documentation and not in the troubleshooting section

Author: Moved in the basic usage at the beginning.
